pytorch_lightning/metrics/classification/helpers.py

```diff
@@ -410,6 +418,9 @@ def _input_format_classification(
     if preds.dtype == torch.float16:
         preds = preds.float()
 
+    # Let threshold default to 0.5 if not set
+    threshold = 0.5 if threshold is None else threshold
```

Comment:
```suggestion
    threshold = threshold or 0.5
```

Reply:
The problem is that if `threshold=0`, this will give 0.5. I know setting it to 0 is silly, but it is supported

Reply:
yeah setting it to 0 won't infer anything, neither 1 I think. Just thinking, should it be updated to allow only (0, 1)??

Reply:
Yeah, why not - it seems reasonable.